test/puppetlabs/puppetdb/cli/services_test.clj

```diff
@@ -546,3 +546,23 @@
                         (set (get-temporal-partitions "reports"))))
                  (is (= (->> event-parts (sort-by :table) (drop 1) set)
                         (set (get-temporal-partitions "resource_events")))))))))))))
+
+(deftest initialize-db
+  (testing "use connection user for database connections"
+    (let [con-mig-user "conn-migration-user-value"
+          mig-pass     "migrator-password-value"
+          user         "user-value"
+          pass         "password-value"
+          validation-fn (fn [config metrics-registry]
+                          (is (= con-mig-user (:connection-migrator-username config)))
+                          (is (= con-mig-user (:user config)))
+                          (is (= mig-pass (:migrator-password config)))
+                          (is (= mig-pass (:password config)))
+                          (throw (Exception.
```

Comment:
I didn't understand why we were throwing the exception at first but I think I get it now. Is this to pop out of the mocked function after we do our assertions and avoid any of the other calls downstream from this which are expecting a connection pool obj?

Reply:
Yes, exactly, we want to stop the execution after we exit our mock `make-connection-pool` because the rest of the code from `init-with-db` expects a valid pooled DB spec map from `make-connection-pool`. If we don't stop the execution with an exception, we would have to mock more methods or the code will try to connect to an actual database.

Reply:
I have added a comment with a short description as to why we have the exception.